crates/network/src/transaction/builder.rs

```diff
@@ -56,6 +56,17 @@ impl<N: Network> TransactionBuilderError<N> {
     }
 }
 
+/// Represents how a transaction handles input/data values
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum TransactionInputKind {
+    /// Only supports 'input' field
+    Input,
```

Comment:
let's make this the default

and also move this type to where we define:

https://github.com/alloy-rs/alloy/blob/54eb6743c2a281cdfc7b989168d83a492aba69da/crates/rpc-types-eth/src/transaction/request.rs#L1135-L1135